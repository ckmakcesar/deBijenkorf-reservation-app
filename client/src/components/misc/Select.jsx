import React from 'react';
import PropTypes from 'prop-types';

import styles from '../../styles/misc/Select.module.css';

const Select = ({
  label,
  options,
  onChange,
  name,
  value,
  prependBlankOption,
  required,
  disabled,
  id,
}) => (
  <label className='input-field-label'>
    <div>{label ? `${label}${label && required ? '*' : ''}` : undefined}</div>

    <select
      {...(id ? { id: `Select-${id}` } : null)} // optional id - should be unique
      className={styles.select}
      name={name}
      value={value}
      onChange={e => { onChange(e.target.value) }}
      required={required}
      disabled={disabled}
    >
      {prependBlankOption ? <option key=''></option> : <></>}
      {options.map((option, index) => (
        <option key={index}>{option.name}</option>
      ))}
    </select>
  </label>
);

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
Select.propTypes = {
  label: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
  })).isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string,
  value: PropTypes.string,
  prependBlankOption: PropTypes.bool,
  required: PropTypes.bool,
  disabled: PropTypes.bool,
  id: PropTypes.string,
};

Select.defaultProps = {
  label: '',
  name: '',
  value: '',
  prependBlankOption: false,
  required: false,
  disabled: false,
  id: '',
};

export default Select;
