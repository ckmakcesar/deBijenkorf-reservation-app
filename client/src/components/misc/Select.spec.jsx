import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Select from './Select';

describe('component:Select', () => {
  const mockLanguageOptions = [
    { name: 'EN' },
    { name: 'ES' },
    { name: 'NL' },
  ];

  it('Should render Select with every optional props absent', () => {
    const wrapper = shallow(
      <Select
        options={mockLanguageOptions}
        onChange={jest.fn()}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render Select with every optional props present', () => {
    const wrapper = shallow(
      <Select
        label='Select a Language'
        options={mockLanguageOptions}
        onChange={jest.fn()}
        name='language-options'
        value={mockLanguageOptions[0].name}
        required={true}
        disabled={false}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
