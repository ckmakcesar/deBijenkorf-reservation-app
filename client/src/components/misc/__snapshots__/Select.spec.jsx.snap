// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:Select Should render Select with every optional props absent 1`] = `
<label
  className="input-field-label"
>
  <div />
  <select
    className="select"
    disabled={false}
    name=""
    onChange={[Function]}
    required={false}
    value=""
  >
    <option
      key="0"
    >
      EN
    </option>
    <option
      key="1"
    >
      ES
    </option>
    <option
      key="2"
    >
      NL
    </option>
  </select>
</label>
`;

exports[`component:Select Should render Select with every optional props present 1`] = `
<label
  className="input-field-label"
>
  <div>
    Select a Language*
  </div>
  <select
    className="select"
    disabled={false}
    name="language-options"
    onChange={[Function]}
    required={true}
    value="EN"
  >
    <option
      key="0"
    >
      EN
    </option>
    <option
      key="1"
    >
      ES
    </option>
    <option
      key="2"
    >
      NL
    </option>
  </select>
</label>
`;
